src/olympia/files/helpers.py

```diff
@@ -78,34 +108,73 @@ def __init__(self, file_obj):
     def __str__(self):
         return str(self.file.id)
 
-    def _extraction_cache_key(self):
-        return ('%s:file-viewer:extraction-in-progress:%s' %
-                (settings.CACHE_PREFIX, self.file.id))
+    def _cache_key(self, key=None):
+        assert key is not None
+        return '{0}:file-viewer:{1}:{2}'.format(
+            settings.CACHE_PREFIX, key, self.file.id
+        )
+
+    @contextlib.contextmanager
+    def lock(self):
+        """Lock the file-viewer.
+
+        If locked, no one is allowed to run any extraction in `self.src` to
+        avoid files / directories get messed up.
+        """
+        cache_key = self._cache_key(LOCKED)
+        msg = Message(cache_key)
+
+        if msg.get():
+            # Already locked
+            yield True
```

Comment:
This value feels reversed to me. I would imagine `lock()` would return `True` if you attained the lock not if it was already locked.

Would be good to at least comment about the return value in the docstring if you'd prefer to keep it this way.